Sources/Yams/Parser.swift

```diff
@@ -108,33 +116,70 @@ public final class Parser {
     /// Constructor.
     public let constructor: Constructor
 
+    /// Encoding
+    public enum Encoding: String {
+        /// Use `YAML_UTF8_ENCODING`
+        case utf8
+        /// Use `YAML_UTF16(BE|LE)_ENCODING`
+        case utf16
+        /// The default encoding, determined at run time based on the String type's native encoding.
+        /// This can be overridden by setting `YAMS_DEFAULT_ENCODING` to either `UTF8` or `UTF16`.
+        /// This value is case insensitive.
+        public static var `default`: Encoding = {
+            let key = "YAMS_DEFAULT_ENCODING"
+            if let yamsEncoding = ProcessInfo.processInfo.environment[key],
+                let encoding = Encoding(rawValue: yamsEncoding.lowercased()) {
+                print("""
+                    `Parser.Encoding.default` was set to `\(encoding)` by the `\(key)` environment variable.
+                    """)
+                return encoding
+            }
+            return "test".utf8.withContiguousStorageIfAvailable({ _ in true }) != nil ? .utf8 : .utf16
```

Comment:
I don't feel strongly, but we could use `key.utf8...` here instead to avoid adding another string to the binary.